### PR TITLE
Fix EXPath HTTP Client socket-closed regression

### PR DESCRIPTION
**Pull request by adamretter**
Closes https://github.com/eXist-db/exist/issues/1709

### CODE REVIEW

File: src/org/exist/xquery/value/BinaryValueFromFile.java [hunk 1]
```diff
@@ -82,6 +92,7 @@ public boolean isClosed() {
     @Override
     public void close() throws IOException {
         channel.close();
+        closeListener.ifPresent(cl -> cl.accept(file));
```
Comment (dizzzz) [inline src/org/exist/xquery/value/BinaryValueFromFile.java]: if ` channel.close();` throws an exception... should we try to execute line 95 as well?

File: extensions/expath/src/org/expath/httpclient/model/exist/EXistResult.java [hunk 2]
```diff
@@ -94,8 +92,13 @@ public void add(final Reader reader) throws HttpClientException {
     @Override
     public void add(final InputStream is) throws HttpClientException {
         try {
-            result.add(BinaryValueFromInputStream.getInstance(context, new Base64BinaryValueType(), is));
-        } catch(final XPathException xpe) {
+            // we have to make a temporary copy of the data stream, as the socket will be closed shortly
+            final TemporaryFileManager temporaryFileManager = TemporaryFileManager.getInstance();
+            final Path tempFile = temporaryFileManager.getTemporaryFile();
```
Comment (dizzzz) [inline extensions/expath/src/org/expath/httpclient/model/exist/EXistResult.java]: Will this for small files, a in-memory operation? I ask for performance reasons.... (IO overhead to disk makes things slow)

Comment (adamretter) [reply extensions/expath/src/org/expath/httpclient/model/exist/EXistResult.java]: I don't think you will notice the difference. As Network is so much slower than disk in the first place

File: src/org/exist/xquery/value/BinaryValueFromFile.java (hunk 1 above)
Comment (adamretter) [reply src/org/exist/xquery/value/BinaryValueFromFile.java]: Good idea. I will fix...

Comment (adamretter) [reply src/org/exist/xquery/value/BinaryValueFromFile.java]: Fixed.

**Comment by joewiz** [comment]
Excellent! Thanks **so** much, Adam and Dannes!

**Pull request merged**